spyder/plugins/run/container.py

```diff
@@ -207,7 +206,7 @@ def run_file(self, selected_uuid=None, selected_executor=None):
 
         first_execution = exec_params['first_execution']
         display_dialog = exec_params['display_dialog']
-        display_dialog = display_dialog or first_execution
+        display_dialog = display_dialog and first_execution
```

Comment:
I am not 100% sure but this was causing the run configuration dialog to open every time when running a new file, which is annoying and makes the tests fail